### PR TITLE
[IMP] mail: chatter attachment upload

### PR DESCRIPTION
**Pull request by tsm-odoo**

### CODE REVIEW

File: addons/mail/static/src/new/views/chatter.js [hunk 1]
```diff
@@ -63,7 +72,16 @@ export class Chatter extends Component {
         );
     }
 
+    get hasAttachmentBox() {
+        return (
+            this.state.isAttachmentBoxOpened ||
+            (this.state.isAttachmentBoxOpened &&
+                (this.attachmentUploader.attachments || this.state.attachments.length > 0))
+        );
```
Comment (alexkuhn) [inline addons/mail/static/src/new/views/chatter.js]: `this.state.isAttachmentBoxOpened ||` is useless

File: addons/mail/static/src/new/views/chatter.js [hunk 2]
```diff
@@ -22,18 +24,25 @@ export class Chatter extends Component {
...
             inChatter: true,
             chatter: {
                 reload: this.load.bind(this),
             },
         });
-        useDropzone(useRef("root"));
+        useDropzone(useRef("root"), (files) => {
+            [...files].forEach(this.attachmentUploader.upload);
+            this.state.isAttachmentBoxOpened = true;
+        });
```
Comment (alexkuhn) [inline addons/mail/static/src/new/views/chatter.js]: It's very unclear what's/when the callback is called without seeing the doc of `useDropzone`.
I feel like dropzone should stay simple and don't have "file" logic.
I would put file logic here explicitly, and have callback for `onDrop` (should see it explicitly named)

**Comment by alexkuhn** [comment]
Note that drag-and-drop only works on attachment box right with this PR.
It's not possible to do on composer for the moment.
I think dropzone code just had a single dropzone for chatter before your PR.
Just wanted to point out the feature of drag/drop in chatter is still incomplete after this PR.

chatter-composer-drag-drop

**Comment by tsm-odoo** [comment]
@alexkuhn Indeed, composer will be done in a following PR to keep things simple to review :+1:

**Pull request merged**